src/Epidemiology/EpiSystem.jl

```diff
@@ -44,45 +45,48 @@ mutable struct EpiSystem{U <: Integer, EE <: AbstractEpiEnv, EL <: EpiList, ER <
-     AbstractEpiEnv,
-    EL <: EpiList, ER <: AbstractTraitRelationship}
-    total_pop = sum(abundances.matrix, dims = 1)[1, :]
-    sorted_grid_ids = sortperm(total_pop, rev = true)
-    sorted_grid_ids = sorted_grid_ids[total_pop[sorted_grid_ids] .> 0]
-    new{U, EE, EL, ER}(abundances, epilist, epienv, ordinariness, relationship, lookup, cache, initial_infected, sorted_grid_ids)
-  end
-  function EpiSystem{U, EE, EL, ER}(abundances::EpiLandscape{U},
-    epilist::EL, epienv::EE, ordinariness::Union{Matrix{Float64}, Missing}, relationship::ER, lookup::EpiLookup, cache::EpiCache, initial_infected::Int64, ordered_active::Vector{Int64}) where {U <: Integer, EE <:
-     AbstractEpiEnv,
-    EL <: EpiList, ER <: AbstractTraitRelationship}
-    new{U, EE, EL, ER}(abundances, epilist, epienv, ordinariness, relationship, lookup, cache, initial_infected, sorted_grid_ids, ordered_active)
-  end
+end
+function EpiSystem(abundances::EpiLandscape{U, VecRNGType}, epilist::EL, epienv::EE,
+    ordinariness::Union{Matrix{Float64}, Missing}, relationship::ER, lookup::EpiLookup,
+    cache::EpiCache, initial_infected::Int64
+    ) where {U <: Integer, VecRNGType <: AbstractVector{<:Random.AbstractRNG},
+    EE <: AbstractEpiEnv, EL <: EpiList, ER <: AbstractTraitRelationship}
+  total_pop = sum(abundances.matrix, dims = 1)[1, :]
+  sorted_grid_ids = sortperm(total_pop, rev = true)
+  sorted_grid_ids = sorted_grid_ids[total_pop[sorted_grid_ids] .> 0]
+  return EpiSystem(abundances, epilist, epienv, ordinariness, relationship, lookup, cache, initial_infected, sorted_grid_ids)
 end
 
 function EpiSystem(popfun::F, epilist::EpiList, epienv::GridEpiEnv,
-    rel::AbstractTraitRelationship, intnum::U; initial_infected = 0) where {F<:Function, U <: Integer}
+      rel::AbstractTraitRelationship, intnum::U; initial_infected = 0,
+      rngtype::Type{R} = Random.MersenneTwister
+      ) where {F<:Function, U <: Integer, R <: Random.AbstractRNG}
 
   # Create matrix landscape of zero abundances
-  ml = emptyepilandscape(epienv, epilist, intnum)
+  ml = emptyepilandscape(epienv, epilist, intnum, rngtype)
+
   # Populate this matrix with species abundances
   popfun(ml, epilist, epienv, rel)
   initial_pop = sum(ml.matrix, dims = 1)
   # Create lookup table of all moves and their probabilities
   home_lookup = genlookups(epienv, epilist.human.movement.home)
   work_lookup = genlookups(epienv, epilist.human.movement.work, initial_pop)
   lookup = EpiLookup(home_lookup, work_lookup)
+  nm = zeros(Float64, size(ml.matrix))
   vm = zeros(Float64, size(ml.matrix))
-  EpiSystem{U, typeof(epienv), typeof(epilist), typeof(rel)}(ml, epilist, epienv, missing, rel, lookup, EpiCache(vm, false), initial_infected)
+  return EpiSystem(ml, epilist, epienv, missing, rel, lookup, EpiCache(nm, vm, false), initial_infected)
```

Comment:
Fix for merge problem:
```suggestion
  return EpiSystem(ml, epilist, epienv, missing, rel, lookup, EpiCache(vm, false), initial_infected)
```